### PR TITLE
[rllib] add augmented random search

### PR DESCRIPTION
**Pull request by eugenevinitsky**
## What do these changes do?

Augmented random search is added as an algorithm: https://arxiv.org/pdf/1803.07055.pdf
Std deviation of meanstdfilter is initialized to 1

### CODE REVIEW

**Review by ericl — COMMENTED** [review]
This looks pretty good, some minor comments.

File: python/ray/rllib/agents/ars/ars.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
+# Code in this file is copied and adapted from
+# https://github.com/openai/evolution-strategies-starter and from
+# https://github.com/modestyachts/ARS
```
[the remark this answers is not in the document]
Comment (eugenevinitsky) [reply python/ray/rllib/agents/ars/ars.py]: Ooh, good catch, thanks!

File: python/ray/rllib/agents/ars/ars.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+        tlogger.record_tabular("UpdateRatio", float(update_ratio))
+
+        tlogger.record_tabular("EpisodesThisIter", noisy_lengths.size)
+        tlogger.record_tabular("EpisodesSoFar", self.episodes_so_far)
+        tlogger.record_tabular("TimestepsThisIter", noisy_lengths.sum())
+        tlogger.record_tabular("TimestepsSoFar", self.timesteps_so_far)
+
+        tlogger.record_tabular("TimeElapsedThisIter", step_tend - step_tstart)
+        tlogger.record_tabular("TimeElapsed", step_tend - self.tstart)
+        tlogger.dump_tabular()
```
[the remark this answers is not in the document]
Comment (eugenevinitsky) [reply python/ray/rllib/agents/ars/ars.py]: I find some of it useful, but I definitely will remove duplicates. Seeing the std deviation of the weights and the grad norm of the update are useful checks that things are going as planned.

**Comment by eugenevinitsky** [comment]
I get: ./yapf.sh: line 51: mapfile: command not found
do you know what this might refer to?

**Comment by ericl** [comment]
Can you put this in the info field of the result instead?

File: python/ray/rllib/agents/ars/ars.py [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+        info = {
+            "weights_norm": np.square(theta).sum(),
+            "grad_norm": np.square(g).sum(),
+            "update_ratio": update_ratio,
+            "episodes_this_iter": noisy_lengths.size,
+            "episodes_so_far": self.episodes_so_far,
+            "timesteps_this_iter": noisy_lengths.sum(),
+            "timesteps_so_far": self.timesteps_so_far,
+            "time_elapsed_this_iter": step_tend - step_tstart,
+            "time_elapsed": step_tend - self.tstart
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/agents/ars/ars.py]: Can you put this in the info field of the result instead?

**Comment by eugenevinitsky** [comment]
Addressed the other comments, still blocked on the yapf thing

**Review by ericl — APPROVED** [review]
Looks good. Did you include the filter change though?

**Comment by eugenevinitsky** [comment]
Yeah, it's changing np.zeros to np.ones in utils.filter

File: python/ray/rllib/utils/filter.py [hunk 4]
```diff
@@ -62,7 +62,7 @@ class RunningStat(object):
     def __init__(self, shape=None):
         self._n = 0
         self._M = np.zeros(shape)
-        self._S = np.zeros(shape)
+        self._S = np.ones(shape)
```
Comment (eugenevinitsky) [inline python/ray/rllib/utils/filter.py]: The relevant filter change, I think.

**Comment by eugenevinitsky** [comment]
Thanks for those fixes btw

**Comment by ericl** [comment]
Going to merge this sans the filter change, since that breaks some of the filter unit tests.

**Comment by akaniklaus** [comment]
Thanks a lot for this. Is there any particular reason why this is not utilizing GPU for computation?

**Pull request merged**